homeassistant/components/remote_calendar/calendar.py

```diff
@@ -0,0 +1,96 @@
+"""Calendar platform for a Remote Calendar."""
+
+from datetime import date, datetime, timedelta
+import logging
+
+from ical.event import Event
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from . import RemoteCalendarConfigEntry
+from .const import CONF_CALENDAR_NAME
+from .coordinator import RemoteCalendarDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: RemoteCalendarConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the remote calendar platform."""
+    coordinator = entry.runtime_data
+    name = entry.data[CONF_CALENDAR_NAME]
+    entity = RemoteCalendarEntity(coordinator, name)
+    async_add_entities([entity])
+
+
+class RemoteCalendarEntity(
+    CoordinatorEntity[RemoteCalendarDataUpdateCoordinator], CalendarEntity
+):
+    """A calendar entity backed by a remote iCalendar url."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: RemoteCalendarDataUpdateCoordinator,
+        name: str,
+    ) -> None:
+        """Initialize RemoteCalendarEntity."""
+        super().__init__(coordinator)
+        self._attr_name = name
+        self._attr_unique_id = name
```

Comment:
My impression from https://developers.home-assistant.io/docs/entity_registry_index#unique-id-requirements is this is not a valid unique id.